### ISSUE TITLE
Breaking change multiplication bool and int

### ISSUE THREAD

**Issue by Tux1** [pandas-dev/pandas#22047]
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
a = pd.DataFrame([[False, True], [False, False]])
print(a.dtypes)
b = a * 1
print(b.dtypes)
```
#### Problem description

With 0.20.3, I get `b` as DataFrame `int32`.
With 0.23.3, I get `b` as DataFrame `object`.

##### With 0.20.3
`a.dtypes =`
0    bool
1    bool
dtype: object

`b.dtypes = `
0    int32
1    int32
dtype: object

##### With 0.23.3
`a.dtypes =`
0    bool
1    bool
dtype: object

`b.dtypes = `
0    object
1    object
dtype: object

#### Versions
pandas 0.20.3 vs pandas 0.23.3

**Comment by jbrockmendel** [comment]
What do you get if you multiply a series instead of a dataframe?

**Comment by jorisvandenbossche** [comment]
A Series works infers the dtype correctly:

```
In [71]: a[0]
Out[71]: 
0    False
1    False
Name: 0, dtype: bool

In [72]: a[0] * 1
Out[72]: 
0    0
1    0
Name: 0, dtype: int64

```

**Comment by Tux1** [comment]
Any idea about the root cause ? If you give me some tips, I can have a look

**Comment by jbrockmendel** [comment]
No idea what changed it, but it'll be fixed by the follow-up after #22074.
